framework/src/main/java/org/checkerframework/framework/util/dependenttypes/DependentTypesHelper.java

```diff
@@ -256,198 +272,86 @@ private void viewpointAdaptExecutable(
 
-        return SystemUtil.mapList(JavaExpression::fromTree, argTrees);
-    }
-
-    /**
-     * Returns true if method is a varargs method or constructor and its varargs arguments are not
-     * passed in an array.
-     *
-     * @param method the method or constructor
-     * @param methodType type of the method or constructor; used for determining the type of the
-     *     varargs formal parameter
-     * @param args the arguments at the call site
-     * @return true if method is a varargs method and its varargs arguments are not passed in an
-     *     array
-     */
-    private boolean isVarArgsInvocation(
-            ExecutableElement method,
-            AnnotatedExecutableType methodType,
-            List<? extends ExpressionTree> args) {
-        if (method != null && method.isVarArgs()) {
-            if (method.getParameters().size() != args.size()) {
-                return true;
-            }
-            AnnotatedTypeMirror lastArg = factory.getAnnotatedType(args.get(args.size() - 1));
-            List<AnnotatedTypeMirror> paramTypes = methodType.getParameterTypes();
-            AnnotatedArrayType lastParam =
-                    (AnnotatedArrayType) paramTypes.get(paramTypes.size() - 1);
-            return lastArg.getKind() != TypeKind.ARRAY
-                    || AnnotatedTypes.getArrayDepth(lastParam)
-                            != AnnotatedTypes.getArrayDepth((AnnotatedArrayType) lastArg);
+        Element ele = TreeUtils.elementFromUse(fieldAccess);
+        if (ele.getKind() != ElementKind.FIELD && ele.getKind() != ElementKind.ENUM_CONSTANT) {
+            return;
         }
-        return false;
+
+        convertAnnotatedTypeMirror(
+                stringExpr ->
+                        StringToJavaExpression.atFieldAccess(
+                                stringExpr, fieldAccess, factory.getChecker()),
+                type);
     }
 
     ///
     /// Standardization
     ///
 
     /**
-     * Creates a TreeAnnotator that standardizes dependent type annotations.
-     *
-     * @return a new TreeAnnotator that standardizes dependent type annotations
-     */
-    public TreeAnnotator createDependentTypesTreeAnnotator() {
-        assert hasDependentAnnotations();
-        return new DependentTypesTreeAnnotator(factory, this);
-    }
-
-    /**
-     * Standardizes the Java expressions in annotations on a constructor invocation.
-     *
-     * @param tree the constructor invocation
-     * @param type the type of the expression; is side-effected by this method
-     */
-    public void standardizeNewClassTree(NewClassTree tree, AnnotatedDeclaredType type) {
-        if (!hasDependentType(type)) {
-            return;
-        }
-
-        TreePath path = factory.getPath(tree);
-        ;
-        parseToPath(path, type);
-    }
-
-    /**
-     * Standardize the Java expressions in annotations in a class declaration.
+     * Viewpoint-adapts the Java expressions in annotations to a class declaration.
```

Comment:
The comment immediately above says this section is about standardization, but here there is more viewpoint-adaptation.